components/clp-py-utils/clp_py_utils/s3_utils.py

```diff
@@ -1,18 +1,74 @@
+import os
 import re
 from pathlib import Path
-from typing import List, Tuple
+from typing import Dict, List, Optional, Tuple, Union
 
 import boto3
 from botocore.config import Config
 from job_orchestration.scheduler.job_config import S3InputConfig
 
-from clp_py_utils.clp_config import S3Config
+from clp_py_utils.clp_config import S3Config, S3Credentials
 from clp_py_utils.compression import FileMetadata
 
 # Constants
 AWS_ENDPOINT = "amazonaws.com"
 
 
+def get_session_credentials(aws_profile: str) -> Optional[S3Credentials]:
+    """
+    Generates AWS credentials created by boto3 when starting a session with given profile.
+    :param aws_profile: Name of profile configured in ~/.aws directory
+    :return: An S3Credentials object with access key pair and session token if applicable.
+    """
+    aws_session = boto3.Session(profile_name=aws_profile)
+    credentials = aws_session.get_credentials()
+    if credentials is None:
+        return None
+
+    return S3Credentials(
+        access_key_id=credentials.access_key,
+        secret_access_key=credentials.secret_key,
+        session_token=credentials.token,
+    )
+
+
+def make_s3_env_vars(config: Union[S3Config, S3InputConfig]) -> Dict[str, str]:
```

Comment:
The method's docstrings need to be updated.

I also don't completely agree with the name of the method and where to put it.
For name, technically this is not s3 env vars, but rather env vars for aws credentials. also feel that the function can simply return a dictionary of only new env-vars, instead of appending to the `os.environ` directly.

Need to think about it for a bit.

Reply:
Can we:
1. Rename this to `get_credential_env_vars`.
2. Remove **os.environ from the return, and let callers combine the return value and **os.environ.

I feel this will make the method clearer